src/client/mod.rs

```diff
@@ -24,39 +24,56 @@ pub mod builder {
 }
 
 
-pub struct Client {
-    transport: ImapTls,
-    state: ClientState,
+pub fn connect(server: &str) -> io::Result<ImapConnectFuture> {
```

Comment:
BTW, curious for any thoughts on this: `connect()` as a free function in the `client` module that returns a `TlsClient`, or `connect()` as a static method on the `TlsClient` itself?

Reply:
I prefer the latter, because then I can write:
```rust
use tokio_imap::TlsClient;
// ...
    let c = TlsClient::connect(...);
// ...
fn foo(c: TlsClient) { ... }
```
instead of
```rust
use tokio_imap::{self, TlsClient};
// ...
    let c = tokio_imap::connect(...);
// ...
fn foo(c: TlsClient) { ... }
```
but it's a pretty minor preference.

Reply:
Yeah, I think I'm back to leaning that way, too.

Reply:
Also favor the second alternative. Even better (for debugging and forward compatibility) would be for `connect` to use a `server` field carried by `TlsClient` rather than a formal parameter.